libsolidity/ast/Types.cpp

```diff
@@ -2335,11 +2335,13 @@ bool StructType::isDynamicallyEncoded() const
 {
 	if (recursive())
 		return true;
-	solAssert(interfaceType(false).get(), "");
-	for (auto t: memoryMemberTypes())
```

Comment:
Also, it was failing here as structs with (nested) mappings are not allowed in memory